### PR TITLE
DFS Traversal of Binary Tree (in C++) and updated README

### PR DESCRIPTION
**Pull request by imsushant12**
I have implemented DFS traversal of Binary Tree with proper description, sample input/output and time/space complexity in C++ under GSSoC'21

Related issue : #248

### CODE REVIEW

File: Tree/readme.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ that, this property should be satisfied at every node in the tree.
 * Sorting the elements of binary search tree and many more
 
 ## Questions :
```
Comment (rish-singhal) [inline Tree/readme.md:49]: again sort the list according to name as earlier.

File: Code/C++/binary_tree_DFS_traversal.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+Enter data to be inserted or type -1 : 50
+Enter left child of 50
+Enter data to be inserted or type -1 : -1
+Enter right child of 50
+Enter data to be inserted or type -1 : -1
+
+DFS Traversal(Pre-Order): 10 20 30 40 50
+DFS Traversal(Post-Order): 30 20 50 40 10
+DFS Traversal(In-Order): 30 20 10 40 50
+
```
Comment (rish-singhal) [inline Code/C++/binary_tree_DFS_traversal.cpp:114]: give one more test case: for this you can truncate previous one.

File: Code/C++/binary_tree_DFS_traversal.cpp [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    cout << "\nDFS Traversal(Post-Order): ";
+    DFS_PostOrder(root);
+    cout << "\nDFS Traversal(In-Order): ";
+    DFS_InOrder(root);
+
+    return 0;
+}
+
+/*
+Time Complexity : O(n)
```
Comment (rish-singhal) [inline Code/C++/binary_tree_DFS_traversal.cpp:84]: specify what is n

File: Code/C++/binary_tree_DFS_traversal.cpp [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+}
+void DFS_InOrder(Node* root)
+{
+    if(root == NULL)
+        return;
+    DFS_InOrder(root->left);
+    cout << root->info << " ";
+    DFS_InOrder(root->right);
+}
+//Driver Program
```
Comment (rish-singhal) [inline Code/C++/binary_tree_DFS_traversal.cpp:67]: add extra lines after each function completion

File: Code/C++/binary_tree_DFS_traversal.cpp [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+void DFS_PreOrder(Node* root)
+{
+    if(root == NULL)
+        return;
+    cout << root->info << " ";
+    DFS_PreOrder(root->left);
+    DFS_PreOrder(root->right);
+
+}
+void DFS_PostOrder(Node* root)
```
Comment (rish-singhal) [inline Code/C++/binary_tree_DFS_traversal.cpp:51]: here too

File: Code/C++/binary_tree_DFS_traversal.cpp [hunk 6]
```diff
@@ -0,0 +1,122 @@
+#include <bits/stdc++.h>
```
Comment (rish-singhal) [inline Code/C++/binary_tree_DFS_traversal.cpp:1]: add information about the program here.

Comment (rish-singhal) [reply Code/C++/binary_tree_DFS_traversal.cpp:1]: as in another PR

File: Code/C++/binary_tree_DFS_traversal.cpp [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+/*
+In the DFS traversal, we start with a node and traverse its adjacent
+child node and continue this process until we are done with all the nodes.
+As trees do not contain cycles,we will surely traverse all the nodes.
+We have tree ways of doing it:
+1.Pre-order way: First print the data, goto left child and then goto right child.
+1.In-order way: First goto left child, print the data and then goto right child.
+1.Post-order way: First goto left child, then goto right child and then print the data.
+*/
+void DFS_PreOrder(Node* root)
```
Comment (rish-singhal) [inline Code/C++/binary_tree_DFS_traversal.cpp:42]: add extra lines here for example

**Review by rish-singhal — CHANGES_REQUESTED** [review]
update these changes then your PR is good to be merged :)

File: Code/C++/binary_tree_DFS_traversal.cpp (hunk 7 above)
Comment (imsushant12) [reply Code/C++/binary_tree_DFS_traversal.cpp:50]: done

File: Code/C++/binary_tree_DFS_traversal.cpp (hunk 6 above)
Comment (imsushant12) [reply Code/C++/binary_tree_DFS_traversal.cpp:7]: done

File: Code/C++/binary_tree_DFS_traversal.cpp (hunk 5 above)
Comment (imsushant12) [reply Code/C++/binary_tree_DFS_traversal.cpp:60]: done

File: Code/C++/binary_tree_DFS_traversal.cpp (hunk 4 above)
Comment (imsushant12) [reply Code/C++/binary_tree_DFS_traversal.cpp:78]: done

File: Code/C++/binary_tree_DFS_traversal.cpp (hunk 3 above)
Comment (imsushant12) [reply Code/C++/binary_tree_DFS_traversal.cpp]: done

File: Code/C++/binary_tree_DFS_traversal.cpp (hunk 2 above)
Comment (imsushant12) [reply Code/C++/binary_tree_DFS_traversal.cpp:124]: done

File: Tree/readme.md (hunk 1 above)
Comment (imsushant12) [reply Tree/readme.md:49]: done sir, please review

**Review by rish-singhal — APPROVED** [review]

**Pull request merged**